internal/provider/common.go

```diff
@@ -0,0 +1,146 @@
+package provider
```

Comment:
This logic moved from `bminventory/inventory.go` 
The only changes are extracting the some logic to functions to make it more clean and easy to read.
new methods are:

1. `isPlatformBM`
2. `isPlatformNone`
3. `isClusterPlatformNone`
4. `isClusterPlatformBM`

Using those methods simplify the already complex cases.